pinot-server/src/main/java/com/linkedin/pinot/server/starter/helix/SegmentMessageHandlerFactory.java

```diff
@@ -34,11 +36,36 @@
   // We only allow one segment refresh/reload happens at the same time
   // The reason for that is segment refresh/reload will temporarily use double-sized memory
   private final Lock _lock = new ReentrantLock();
+  private final Semaphore _refreshThreadSemaphore;
 
   private final SegmentFetcherAndLoader _fetcherAndLoader;
 
-  public SegmentMessageHandlerFactory(SegmentFetcherAndLoader fetcherAndLoader) {
+  public SegmentMessageHandlerFactory(SegmentFetcherAndLoader fetcherAndLoader, DataManager dataManager) {
```

Comment:
Does this class need dataManager, or can we just pass maxParallelRefreshThreads to the constructor?

Reply:
The class does not need DataManager. However, I prefer passing DataManager in, so other configs can be retrieved if we need to do so. Otherwise, the number of arguments keep increasing over time, a la RealtimeSegmentImpl (that I re-factored a few weeks back to reduce the number of args)